helpers.c

```diff
@@ -26,9 +26,19 @@ int sync_file_range(int fd, uint64_t offset, uint64_t nbytes,
 }
 #endif
 
+#if defined (__APPLE__)
```

Comment:
You might be able to move this into `os/os-mac.h` by doing `#define CONFIG_POSIX_FADVISE` over there and providing an implementation in that file...